### PR TITLE
config: move conmon specific fields in to runtime handler

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind deprecation
#### What this PR does / why we need it:
This makes the transition to conmonrs easier, as we can update conmonrs fields inpendently of the conmon ones.
#### Does this PR introduce a user-facing change?

```release-note
Deprecate `conmon`, `conmon-cgroup`, and `conmon-env` config options in favor of runtime handler specific fields `monitor-path`, `monitor-cgroup`, and `monitor-env`
```

### CODE REVIEW

**Comment by haircommander** [comment]
good point, I've updated the kind so it'll be correctly catagorized

**Comment by haircommander** [comment]
interestingly, I think some of the tests will be helped by https://github.com/cri-o/cri-o/pull/5740

/retest

File: pkg/config/template.go [hunk 1]
```diff
@@ -1103,6 +1106,12 @@ const templateStringCrioRuntimeRuntimesRuntimeHandler = `# The "crio.runtime.run
 {{ $.Comment }}allowed_annotations = [
 {{ range $opt := $runtime_handler.AllowedAnnotations }}{{ $.Comment }}{{ printf "\t%q,\n" $opt }}{{ end }}{{ $.Comment }}]
 {{ $.Comment }}{{ end }}
+{{ $.Comment }}monitor_path = "{{ $runtime_handler.MonitorPath }}"
+{{ $.Comment }}{{ if $runtime_handler.MonitorEnv }}
+{{ $.Comment }}monitor_env = [
+{{ range $opt := $runtime_handler.MonitorEnv }}{{ $.Comment }}{{ printf "\t%q,\n" $opt }}{{ end }}{{ $.Comment }}]
+{{ $.Comment }}{{ end }}
+{{ $.Comment }}monitor_cgroup = "{{ $runtime_handler.MonitorCgroup }}"
```
Comment (donpenney) [inline pkg/config/template.go:1114]: The new option names are missing the runtime_ prefix to align with the declarations in config.go

File: pkg/config/config.go [hunk 2]
```diff
@@ -1111,6 +1109,11 @@ func (c *RuntimeConfig) ValidateRuntimes() error {
 			logrus.Warnf("'%s is being ignored due to: %q", name, err)
 			failedValidation = append(failedValidation, name)
 		}
+		if c.DefaultRuntime == name {
+			if err := c.TranslateMonitorFields(handler); err != nil {
```
Comment (donpenney) [inline pkg/config/config.go:1113]: It looks like there's no default set for MonitorPath, as defaultRuntimeHandler doesn't set one and TranslateMonitorFields is only called for c.DefaultRuntime. Without runtime_monitor_path explicitly set in the config file, I think this means MonitorPath will be an empty string for any non-default runtime handlers.

File: pkg/config/template.go (hunk 1 above)
Comment (haircommander) [reply pkg/config/template.go:1115]: we already have privileged_without_host_devices not prefixed. I personally think it's alright, as these fields are configuring the monitor, not the runtime

File: pkg/config/config.go (hunk 2 above)
Comment (haircommander) [reply pkg/config/config.go:1115]: good point! I've updated to run if the runtime is of type `oci`

File: pkg/config/template.go (hunk 1 above)
Comment (donpenney) [reply pkg/config/template.go:1115]: Aren't the new options defined with a runtime_ prefix in config.go, though? ie. runtime_monitor_path? So the example config this template generates would be missing that prefix and not be parsed?

Comment (haircommander) [reply pkg/config/template.go:1115]: oh jeez, you're right. I'll fix it there

Comment (haircommander) [reply pkg/config/template.go:1115]: fixed

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**